pkg/sources/circleci/circleci.go

```diff
@@ -251,18 +324,55 @@ func (s *Source) chunkAction(ctx context.Context, proj project, bld build, act a
 						VcsType:     proj.VCS,
 						Username:    proj.Username,
 						Repository:  proj.RepoName,
-						BuildNumber: int64(bld.BuildNum),
+						BuildNumber: int64(projectBuild.BuildNum),
 						BuildStep:   stepName,
 						Link:        linkURL,
 					},
 				},
 			},
 			Verify: s.verify,
 		}
-		chunk.Data = data.Bytes()
 		if err := data.Error(); err != nil {
 			return err
 		}
+
+		if err := common.CancellableWrite(ctx, chunksChan, chunk); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *Source) chunkCircleCIYamlString(ctx context.Context, proj project, projectBuild build, yamlString string, chunksChan chan *sources.Chunk) error {
```

Comment:
This method is substantially similar to `chunkAction`; I think it's probably worth factoring out the logic that fetches the output of an action into its own method, and then unifying the functionality.